icu4c/source/tools/toolutil/unewdata.cpp

```diff
@@ -57,14 +57,24 @@ udata_create(const char *dir, const char *type, const char *name,
         return NULL;
     }
     
+    char dirSepChar = U_FILE_SEP_CHAR;
+#if (U_FILE_SEP_CHAR != U_FILE_ALT_SEP_CHAR)
+    // We may need to append a different directory separator when building for Cygwin or MSYS2.
+    if(dir != NULL  && *dir !=0) {
+      if(strchr(dir, U_FILE_SEP_CHAR) == NULL && strchr(dir, U_FILE_ALT_SEP_CHAR) != NULL) {
```

Comment:
Feedback from Markus was that we should be using the `uprv_strstr` instead of directly using `strchr`.

Additionally, he suggested that in new code `nullptr` should be used instead of `NULL` (though it looks like this file uses `NULL` everywhere, so this is more of a nit) .